lib/api/apiUtils/object/objectLockHelpers.js

```diff
@@ -123,10 +123,13 @@ function isObjectLocked(bucket, objectMD, headers) {
         if (objectLegalHold) {
             return true;
         }
-        const objectRetention = objectMD.retentionInfo;
+        let objectRetention = objectMD.retentionInfo;
         if (!objectRetention) {
             return false;
         }
+        if (!objectRetention.retention) {
+            objectRetention = { retention: objectRetention };
```

Comment:
Isn't this a circular reference?
```js
let objectRetention = objectMD.retentionInfo;
objectRetention = { retention: objectRetention };
```
Also, `retention` is not a property in ObjectMD as I see here https://github.com/scality/Arsenal/blob/development/7.7/lib/models/ObjectMD.js#L911

Reply:
resolved with the great flattening https://github.com/scality/cloudserver/pull/2794